### ISSUE TITLE
Poor Training Results

### ISSUE THREAD

**Issue by JunhaoZhu0220** [hp-l33/AiM#15]
I have trained the AiM-B model with the provided first stage model on 4 A800 with following configs:
`--batch-size 256 --lr 4e-4 --epochs 300` with `d_model=768, n_layer=24, adaln_group=False` for AiM-B, and other parameters are set to default values. 
The loss curve for around first 200 epochs are provided here: https://api.wandb.ai/links/ml-nyush/9dh8lq2l , which is lower than 7.

I generated 50k samples with `temperature=1.0, top_p=1.0, top_k=0, cfg_scale=2.0` and calculated the fid with the 50k validation set of imagenet, the result is 28.6272, which is much higher than expected. Some generated samples are also provided below.

I want to ask that which configs may lead to the poor training result, and why the adaln_group is set to `false` instead of 24 groups in the paper?

Class 0:
![Image](https://github.com/user-attachments/assets/68649e91-e130-420f-a1e2-19453d4b293c)

Class 6:
![Image](https://github.com/user-attachments/assets/cc21bc9a-fc17-4a93-8b22-c948d12e1e1c)

Class 250:
![Image](https://github.com/user-attachments/assets/ba2b8510-1349-4c04-9105-24e2441ae344)

**Comment by hp-l33** [comment]
The loss curve you provided looks really strange. In my experiment, the loss of aim-b has never been lower than 7, just like the loss curve I provided in the README.

**Comment by hp-l33** [comment]
For aim-b, adaln_group=false is equivalent to group=24.

**Comment by hp-l33** [comment]
Could you please check the following:
 
1. Whether the VQVAE can reconstruct images normally? If you are using a pre-encoded dataset, please check whether the npy files can be reconstructed back into images.
2. Could you try training aim-l and observe the loss? If your computing resources do not support it, could you try loading the pre-trained weights of aim-b and fine-tuning it?

I'm very curious as to why different people have obtained completely different results when trying to reproduce my work. ControlAR conducted experiments based on our work, and it can work properly. However, some people have encountered problems similar to yours (the loss is abnormally low). I hope to find out the specific reasons.

**Comment by JunhaoZhu0220** [comment]
For the pre-encoding process, I used the extract_codes_c2i.py script from llamagen. For the parameters, I set the `image-size` to 256, `codebook-size` to 16384, `codebook-embed-dim` to 8, and did not use `ten-crop` so it will only generate two samples per image: the encoded sample for the original image and the encoded sample for the flipped image. The following is the reconstructed image.

![Image](https://github.com/user-attachments/assets/d7d062f5-c07b-415c-9522-f7e9fce2a4e0)

**Comment by hp-l33** [comment]
Can you generate reasonable images using the pre-trained weights provided in the repo? And how do you load the weights that you trained yourself?

**Comment by JunhaoZhu0220** [comment]
For the pre-trained weight provided in the repo, I evaluated it with torch-fidelity package, and the fid is 5.5730.
To load my trained weight, I used `model.load_state_dict(torch.load(args.aim_ckpt, map_location='cpu'), strict=False)` where `model = AiM_models["AiM-B"]()`, and pass the directory of the folder containing the AiM-B checkpoint to `args.aim_ckpt`.

**Comment by hp-l33** [comment]
1. I'm not sure if it's reasonable to use the torch-fidelity package to evaluate the FID of ImageNet, because I haven't seen any official repository of similar works using this method. I strongly recommend that you use the widely used ADM evaluation script (https://github.com/openai/guided-diffusion.git). If you don't want to use TensorFlow, then you can also try the mar evaluation script. As for the sample script, you can refer to the one provided in the repository of my other work: https://github.com/hp-l33/ARPG/blob/main/sample_c2i_ddp.py.
2. Have you tried setting ``strict=True`` and successfully read the ckpt?
3. I really don't know why your loss converges extremely quickly and there is a situation where the inference result doesn't match the loss. In my experiment, when the loss of aim-b is around 7.5, the convergence speed is already very slow, and it takes more than nearly half of the total training epochs to converge to 7.4. I don't think this is a problem with the code I released, because we directly used this repository to supplement experiments on different devices during the rebuttal stage of this work in November last year, and it worked well.
4. This is just my guess. It is highly likely to be a problem with the data. When we used the pre-encoded data, we performed ``ten-crop`` (I'm not sure if its impact is that significant), and when not using pre-encoding, we applied random crop (as provided in the repository).
5. If you like, you can add my WeChat @hp_l33 for real-time communication.  I'm more than happy to help you answer any questions regarding this work.

**Comment by JunhaoZhu0220** [comment]
Thank you for your response! I will take a look at it.
